ledger-core/virtual/integration/outgoing_test.go

```diff
@@ -105,52 +157,68 @@ func TestVirtual_CallMethodOutgoing_WithPulseChange(t *testing.T) {
-			assert.Equal(t, firstPulse, report.OrderedPendingEarliestPulse)
-			assert.Equal(t, int32(1), report.OrderedPendingCount)
+			// check for pending counts must be in tests: call constructor/call terminal method
+			assert.Equal(t, objectAGlobal, report.Object)
+			assert.Zero(t, report.DelegationSpec)
 			return false
 		})
 		typedChecker.VDelegatedCallRequest.Set(func(request *payload.VDelegatedCallRequest) bool {
-			assert.Zero(t, request.DelegationSpec)
+			assert.Equal(t, objectAGlobal, request.Callee)
+
+			msg := payload.VDelegatedCallResponse{
+				Callee: request.Callee,
+				ResponseDelegationSpec: payload.CallDelegationToken{
+					TokenTypeAndFlags: payload.DelegationTokenTypeCall,
+					PulseNumber:       server.GetPulse().PulseNumber,
+					Callee:            request.Callee,
+					Outgoing:          request.CallOutgoing,
+					DelegateTo:        server.JetCoordinatorMock.Me(),
+				},
+			}
 
-			server.SendPayload(ctx, &payload.VDelegatedCallResponse{
-				Callee:                 request.Callee,
-				ResponseDelegationSpec: expectedToken,
-			})
+			if firstVDelegatedCallRequest {
+				assert.Zero(t, request.DelegationSpec)
+				msg.ResponseDelegationSpec.Approver = firstApprover
+				firstVDelegatedCallRequest = false
+			} else {
+				assert.Equal(t, firstExpectedToken, request.DelegationSpec)
+				msg.ResponseDelegationSpec.Approver = secondApprover
+			}
 
+			server.SendPayload(ctx, &msg)
 			return false
 		})
 		typedChecker.VDelegatedRequestFinished.Set(func(finished *payload.VDelegatedRequestFinished) bool {
-			assert.Equal(t, expectedToken, finished.DelegationSpec)
+			assert.Equal(t, objectAGlobal, finished.Callee)
+			assert.Equal(t, secondExpectedToken, finished.DelegationSpec)
 			return false
 		})
 		typedChecker.VCallRequest.Set(func(request *payload.VCallRequest) bool {
-			assert.Equal(t, firstPulse, request.CallOutgoing.Pulse())
-			if callCount == 0 {
-				assert.Zero(t, request.DelegationSpec)
+			assert.Equal(t, objectBGlobal, request.Callee)
+
+			if firstVCallRequest {
+				assert.Equal(t, secondPulse, request.CallOutgoing.Pulse()) // new pulse
+				assert.Equal(t, firstExpectedToken, request.DelegationSpec)
+				assert.NotEqual(t, payload.RepeatedCall, request.CallRequestFlags.GetRepeatedCall())
+
 				server.IncrementPulseAndWaitIdle(ctx)
-				callCount++
+				thirdPulse = server.GetPulse().PulseNumber
+				secondExpectedToken.PulseNumber = thirdPulse
+				expectedVCallRequest.DelegationSpec = secondExpectedToken
+				firstVCallRequest = false
 				// request will be sent in previous pulse
 				// omit sending
 				return false
 			}
 
-			assert.Equal(t, payload.RepeatedCall, request.CallRequestFlags.GetRepeatedCall())
+			assert.Equal(t, secondPulse, request.CallOutgoing.Pulse()) // the same pulse
 
-			assert.Equal(t, expectedToken, request.DelegationSpec)
+			// reRequest -> check all fields
+			expectedVCallRequest.CallOutgoing = request.CallOutgoing
```

Comment:
why you disabled field from check?

Reply:
I can only check pulse and I've done it above.